lib/pf/role/pool.pm

```diff
@@ -0,0 +1,179 @@
+package pf::role::pool;
+
+=head1 NAME
+
+pf::role::pool - Object oriented module for VLAN pool
+
+=head1 SYNOPSIS
+
+The pf::role::pool module contains the functions necessary for the VLAN Pool.
+
+=cut
+
+use strict;
+use warnings;
+
+use Log::Log4perl;
+
+use pf::config;
+use pf::util qw (%Config);
+use pf::log();
+
+use pf::node;
+
+use Number::Range;
+
+=head1 SUBROUTINES
+
+Warning: The list of subroutine is incomplete
+
+=cut
+
+=head2 new
+
+Constructor.
+
+=cut
+
+sub new {
+    my $logger =  pf::log::get_logger();
+    $logger->debug("instantiating new pf::role::pool object");
+    my ( $class, %argv ) = @_;
+    my $this = bless {}, $class;
+    return $this;
+}
+
+sub getVlanFromPool {
+    my ($self, $args) = @_;
+    my $logger =  pf::log::get_logger();
+
+    return $pool_args->{'vlan'} if $self->rangeValidator($args->{'vlan'});
+    my $range = Number::Range->new($args->{'vlan'});
+    my $vlan;
+    if ($Config{advanced}{vlan_pool_technique} eq "username_hash") {
+        $vlan = $self->getVlanByUsername($args, $range);
+    } else {
+        $vlan = $self->getRoundRobin($args, $range);
+    }
+    return $vlan;
+}
+
+=head2 rangeValidator
+
+Validate the range definition
+Should be something like that 20..23 or 20..23,27..30
+
+=cut
+
+sub rangeValidator {
+    my ($self, $range) =@_;
+    my $rangesep = qr/(?:\.\.)/;
+    my $sectsep  = qr/(?:\s|,)/;
+    my $validation = qr/(?:
+         [^0-9,. -]|
+         $rangesep$sectsep|
+         $sectsep$rangesep|
+         \d-\d|
+         ^$sectsep|
+         ^$rangesep|
+         $sectsep$|
+         $rangesep$
+         )/x;
+    return 1 if ($range =~ m/$validation/g);
+    return 0;
+}
+
+
+=head2 getRoundRobin
+
+Return the vlan id based on the last registered device + 1
+First test if the last_vlan of the device is in the range then use it
+Else get the last registered device vlan and add + 1 (+1 in the range)
+
+=cut
+
+sub getRoundRobin {
+    my ($self, $args, $range) = @_;
+    my $logger =  pf::log::get_logger();
+
+    my $vlan_count = $range->size;
+    my $node_info_complete = node_view($args->{'mac'});
+    if ( defined($node_info_complete->{'last_vlan'}) && $range->inrange($node_info_complete->{'last_vlan'}) ) {
+        $logger->debug("NODE LAST VLAN ".$node_info_complete->{'last_vlan'});
```

Comment:
Clarify log statement with something like:
"Using the VLAN that was last assigned to the node: ". $node_info_complete->{'last_vlan'}